src/commands/test.js

```diff
@@ -0,0 +1,498 @@
+        skipWaitForEthereum,
+        skipWaitForIpfs,
+        skipWaitForPostgres,
+      })
+    } catch {
+      await stopTestEnvironment(toolbox, tempdir)
+      process.exitCode = 1
+      return
+    }
+
+    // Bring up Graph Node separately, if a standalone node is used
+    let nodeProcess
+    let nodeOutputChunks = []
+    if (standaloneNode) {
+      try {
+        nodeProcess = await startGraphNode(
+          toolbox,
+          standaloneNode,
+          standaloneNodeArgs,
+          nodeOutputChunks,
+        )
+      } catch {
+        toolbox.print.error('')
+        toolbox.print.error('  Graph Node')
+        toolbox.print.error('  ----------')
+        toolbox.print.error(
+          indent('  ', Buffer.concat(nodeOutputChunks).toString('utf-8')),
+        )
+        toolbox.print.error('')
+        await stopTestEnvironment(toolbox, tempdir)
+        process.exitCode = 1
+        return
+      }
+    }
+
+    // Wait for Graph Node to come up
+    try {
+      await waitForGraphNode(toolbox)
+    } catch {
+      toolbox.print.error('')
+      toolbox.print.error('  Graph Node')
+      toolbox.print.error('  ----------')
+      toolbox.print.error(
+        indent(
+          '  ',
+          await graphNodeLogs(toolbox, tempdir, standaloneNode, nodeOutputChunks),
+        ),
+      )
+      toolbox.print.error('')
+      await stopTestEnvironment(toolbox, tempdir)
+      process.exitCode = 1
+      return
+    }
+
+    // Run tests
+    let result = await runTests(toolbox, tempdir, testCommand)
+
+    // Bring down Graph Node, if a standalone node is used
+    if (nodeProcess) {
+      try {
+        await stopGraphNode(toolbox, nodeProcess)
+      } catch {
+        // do nothing (the spinner already logs the problem)
+      }
+    }
+
+    // Bring down the test environment
+    try {
+      await stopTestEnvironment(toolbox, tempdir)
+    } catch {
+      // do nothing (the spinner already logs the problem)
+    }
+
+    if (result.exitCode == 0) {
+      toolbox.print.success('✔ Tests passed')
+    } else {
+      toolbox.print.error('✖ Tests failed')
+    }
+
+    // Always print the test output
+    toolbox.print.info('')
+    toolbox.print.info('  Output')
+    toolbox.print.info('  ------')
+    toolbox.print.info('')
+    toolbox.print.info(indent('  ', result.stdout))
+
+    if (result.exitCode !== 0) {
+      // If there was an error, print the error output as well
+      toolbox.print.error('')
+      toolbox.print.error('  Errors')
+      toolbox.print.error('  ------')
+      toolbox.print.error('')
+      toolbox.print.error(indent('  ', result.stderr))
+    }
+
+    if (result.exitCode !== 0 || nodeLogs) {
+      toolbox.print.info('')
+      toolbox.print.info('  Graph Node')
+      toolbox.print.info('  ----------')
+      toolbox.print.info('')
+      toolbox.print.info(
+        indent(
+          '  ',
+          await graphNodeLogs(toolbox, tempdir, standaloneNode, nodeOutputChunks),
```

Comment:
This is actually where mine failed. I believe what happened was because there was no graph no logs since graph node didn't start up, it didn't give any message and it errored out

Reply:
Here is the whole output:
```
✖ Tests failed

  Output
  ------


  Compiling your contracts...
  ===========================
  > Compiling ./contracts/Gravity.sol
  > Compiling ./contracts/Migrations.sol



    Contract: Basic event handlers
      1) "before all" hook


    0 passing (6s)
    1 failing

    1) Contract: Basic event handlers
         "before all" hook:
       Uncaught FetchError: request to http://localhost:8030/graphql failed, reason: socket hang up
        at ClientRequest.<anonymous> (node_modules/cross-fetch/node_modules/node-fetch/index.js:133:11)
        at Socket.socketOnEnd (_http_client.js:440:9)
        at endReadableNT (_stream_readable.js:1183:12)
        at processTicksAndRejections (internal/process/task_queues.js:80:21)





  Errors
  ------



  Graph Node
  ----------
```

I think there could be another line added that supplements `Output`, `Errors`, `Graph Node` , but isn't dependant on results. But I will leave it up to you to decide

Reply:
Let's retry this based on the latest version of this branch. It should give us more useful logs now.